### PR TITLE
graduate MinDomainsInPodTopologySpread to stable

### PR DESCRIPTION
**Pull request by sanposhiho**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

graduate MinDomainsInPodTopologySpread to stable.

#### Which issue(s) this PR fixes:

Part of https://github.com/kubernetes/enhancements/issues/3022

#### Does this PR introduce a user-facing change?

```release-note
Graduated support for `minDomains` in pod topology spread constraints, to general availability.
The `MinDomainsInPodTopologySpread` feature gate no longer has any effect, and the field is
always available within the Pod and PodTemplate APIs.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/issues/3022
```

### CODE REVIEW

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -1052,7 +1053,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 
 	MemoryQoS: {Default: false, PreRelease: featuregate.Alpha},
 
-	MinDomainsInPodTopologySpread: {Default: true, PreRelease: featuregate.Beta},
+	MinDomainsInPodTopologySpread: {Default: true, PreRelease: featuregate.GA},
```
Comment (sanposhiho) [inline pkg/features/kube_features.go:1056]: As suggested here, it keeps the feature gate itself while removing all the references to this gate.
https://github.com/kubernetes/enhancements/pull/4407#issuecomment-1927256512

**Comment by sftim** [comment]
Changelog suggestion
```diff
-graduate MinDomainsInPodTopologySpread to stable
+Graduated support for `minDomains` in pod topology spread constraints, to general availability.
+The `MinDomainsInPodTopologySpread` feature gate no longer has any effect, and the field is
+always available within the Pod and PodTemplate APIs.
```

File: pkg/features/kube_features.go (hunk 1 above)
Comment (alculquicondor) [inline pkg/features/kube_features.go:1056]:
```suggestion
	MinDomainsInPodTopologySpread: {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.32
```

File: pkg/scheduler/framework/plugins/podtopologyspread/common.go [hunk 2]
```diff
@@ -116,7 +116,7 @@ func (pl *PodTopologySpread) filterTopologySpreadConstraints(constraints []v1.To
 				NodeAffinityPolicy: v1.NodeInclusionPolicyHonor,  // If NodeAffinityPolicy is nil, we treat NodeAffinityPolicy as "Honor".
```
Comment (alculquicondor) [inline pkg/scheduler/framework/plugins/podtopologyspread/common.go:116]: above:

```golang
MinDomains: ptr.Deref(c.MindDomains, 1)
```

**Review by alculquicondor — COMMENTED** [review]
/approve

**Comment by liggitt** [comment]
mechanics look good, will wait for @alculquicondor to lgtm from sig / graduation requirements perspective

**Pull request merged**